src/pages/plugins-2023/index.md.ejs

```diff
@@ -12,66 +12,63 @@ Some plugins are maintained in-house, while others are provided by the community
 On this page, we provide an `npm` link and total downloads per month for any plugins
 we find or publish ourselves, so you only need to look in one place.
 
-<!-- Plugin groups -->
-<%Object.keys(allPlugins).forEach(function (group) {%>
-### Group name: <%=group%>
-Group description: <%=allPlugins[group].XdescX%>. This will be a longer description eventually, giving background on what this family of plugins could be used for, and how to use them. Terminology should be decided between group/family/clan/etc
-
-<%Object.keys(allPlugins[group]).forEach(function (pluginName) {
-  if('XdescX' == pluginName) {return}
-  let plugin = allPlugins[group][pluginName]%>
-- <%=plugin.title%>
-  - <%=plugin.desc%>. This will be a short description taken from the package.json file.
-
-<%  })%>
-<%})%>
-
-<!-- Plugin groups v2 -->
+Table of contents:
+- This intro about the plugins
+- Block list of families with description, link to full description block
+- Each family name with full description, followed by block-list of each plugin in family (with short description)
+- Subheadings of each family with short description
+- Each plugin with badges and description (links active)
+- 'Got a plugin to share?' section
+
+<!-- List of families -->
+## Plugin Families
+*Click to see further description and plugin list*
 <%
-let groupNav = '<div>'
+let familyList = `<div style='display:flex; flex-flow:row wrap;'>`
 Object.keys(allPlugins).forEach(function (group) {
-  groupNav += '<table><tr style="background-color:#f0f0f0"><td style="background-color:#505050">'
-  groupNav += '<h4>'+group+'</h4><p>'+allPlugins[group].XdescX+' This will be a longer description eventually, giving background on what this family of plugins could be used for, and how to use them. Terminology should be decided between group/family/clan/etc.</p></td><td style="background-color:#909090"><dl>'
-  Object.keys(allPlugins[group]).forEach(function (pluginName) {
-    if('XdescX' == pluginName) {return}
-    let plugin = allPlugins[group][pluginName]
-    groupNav += '<dt>'+plugin.title+'</dt><dd>'+plugin.desc+' This will be a short description taken from the package.json file.</dd>'
-  })
-  groupNav += '</dl></td></tr></table>'
+  let family = allPlugins[group]
+  familyList += `
+  <div style='flex: 0 0 49%; margin-bottom: 1%; margin-right: 1%; padding: 0.5% 0.5%; background-color: #dcf1f3; border: 1px solid #dcf1f3; border-radius: 3px; box-shadow: none;'>
+    <div style='font-size:90%; color: #379199; text-transform: uppercase;'><a href='#family_${group}'>${group}</a></div>
+    <div style='font-size:80%'>${family.XdescX}. Longer description to be added to plugins_def file.</div>
+  </div>
+  `
 })
-groupNav += '</div>'
+familyList += '</div>'
```

Comment:
this is sort of defeating the purpose of using a template language like ejs
do this instead:
<% for(...) { %>
   <!-- normal html -->
  <div>...</div> 
<% } %>

Reply:
I'm not sure I understand what can be changed here - there is a for loop on line 28 already.

I understand it's definitely over-complex, but in order to get the layout customisation that markdown lacks and html gives, the ejs unescaped value tag <%-html_variable%> must be used - thankfully this can be wrapped in a standard for loop (L28) so that I can loop through the family names in order to add them in where required.

Reply:
something like this:
```
<%
let familyList = `<div style='display:flex; flex-flow:row wrap;'>`
Object.keys(allPlugins).forEach(function (group) {
  let family = allPlugins[group]
%>
  <div style='flex: 0 0 49%; margin-bottom: 1%; margin-right: 1%; padding: 0.5% 0.5%; background-color: #dcf1f3; border: 1px solid #dcf1f3; border-radius: 3px; box-shadow: none;'>
    <div style='font-size:90%; color: #379199; text-transform: uppercase;'><a href='#family_<%=group%>'><%=group%></a></div>
    <div style='font-size:80%'><%=family.XdescX%>. Longer description to be added to plugins_def file.</div>
  </div>
<% }) %>
```

Reply:
Ah - I was under the impression that html use was discouraged in markdown. I'll take it out of the ejs variable and put it in the file directly